app/scripts/services/journalUploadService.js

```diff
@@ -0,0 +1,139 @@
+angular
+  .module('confRegistrationWebApp')
+  .service('journalUploadService', function JournalUploadService(
+    $rootScope,
+    $http,
+    $q,
+    modalMessage,
+  ) {
+    const path = function(id) {
+      return `conferences/${id}`;
+    };
+
+    this.getRegistrationData = (
+      conferenceId,
+      queryParameters = {
+        page: 1,
+        limit: 20,
+        orderBy: 'last_name',
+        order: 'ASC',
+        filter: '',
+        filterAccountTransferErrors: 'yes',
+        filterAccountTransfersByExpenseType: '',
+        filterAccountTransfersByPaymentType: '',
+        filterPayment: '',
+        filterRegType: '',
+        includeAccountTransfers: true,
+        includeCheckedin: 'yes',
+        includeWithdrawn: 'yes',
+        includeIncomplete: 'yes',
+        primaryRegistrantOnly: true,
+      },
+    ) => {
+      const defer = $q.defer();
```

Comment:
You shouldn't need to use `defer`, just return the promise and return values/rethrow errors in the promise then/catch.

Applies to other lines too.

Reply:
Gotcha, it's interesting that there's so many different ways it's done in this codebase haha